### PR TITLE
[ARIES-1886] New properties for tuning blueprint deployment

### PR DESCRIPTION
**Pull request by philippberger**
VM property "org.apache.aries.blueprint.default.timeout" can be used to
define the default time out.

VM property "org.apache.aries.blueprint.extender.threads" can be used to
define the count of threads used for deployment. The default value is
changed from 3 to 10 threads.

### CODE REVIEW

File: blueprint/blueprint-core/src/main/java/org/apache/aries/blueprint/container/BlueprintContainerImpl.java [hunk 1]
```diff
@@ -210,6 +212,17 @@ public BlueprintListener getEventDispatcher() {
         return eventDispatcher;
     }
 
+    private long getDefaultTimeout() {
+        long timeout = 5 * 60 * 1000;
+        try {
+            timeout = Long.getLong(DEFAULT_TIMEOUT_PROPERTY, timeout);
```
Comment (cschneider) [inline blueprint/blueprint-core/src/main/java/org/apache/aries/blueprint/container/BlueprintContainerImpl.java]: I propose to log the actual timeout used if it differs from the default.

File: blueprint/blueprint-core/src/main/java/org/apache/aries/blueprint/container/BlueprintExtender.java [hunk 2]
```diff
@@ -95,7 +96,14 @@ public void start(BundleContext ctx) {
         handlers = new NamespaceHandlerRegistryImpl(trackingContext);
         executors = new ScheduledExecutorServiceWrapper(ctx, "Blueprint Extender", new ScheduledExecutorServiceFactory() {
             public ScheduledExecutorService create(String name) {
-                return Executors.newScheduledThreadPool(3, new BlueprintThreadFactory(name));
+                int extenderThreads = 10;
```
Comment (cschneider) [inline blueprint/blueprint-core/src/main/java/org/apache/aries/blueprint/container/BlueprintExtender.java]: We should keep the default of 3. So people who do not change it do not see a different behaviour.

File: blueprint/blueprint-core/src/main/java/org/apache/aries/blueprint/container/BlueprintExtender.java [hunk 3]
```diff
@@ -95,7 +96,14 @@ public void start(BundleContext ctx) {
...
+                try {
+                    extenderThreads = Integer.getInteger(EXTENDER_THREADS_PROPERTY, extenderThreads);
```
Comment (cschneider) [inline blueprint/blueprint-core/src/main/java/org/apache/aries/blueprint/container/BlueprintExtender.java]: Like for the timeout the number of threads should be logged if it was changed by the user.

**Review by cschneider — CHANGES_REQUESTED** [review]
I am in favour of this change. I noted some small improvements below.

**Review by cschneider — APPROVED** [review]
Looks good now.
We are currently moving this repo to git. If it is ok I will wait a few days for the migration to succeed.
On git the merge will be simpler and retain your authorship.

**Comment by philippberger** [comment]
> Looks good now.
> We are currently moving this repo to git. If it is ok I will wait a few days for the migration to succeed.
> On git the merge will be simpler and retain your authorship.

Sure, that's okay.
Thank you!

**Comment by cschneider** [comment]
I just applied the PR to trunk. Can you check and close this PR if it is fine now?

**Comment by philippberger** [comment]
Looks good, thanks.

**Pull request closed without merging**